qos-monitor/src/main/java/eu/arrowhead/core/qos/service/ping/monitor/PingEventProcessor.java

```diff
@@ -0,0 +1,146 @@
+package eu.arrowhead.core.qos.service.ping.monitor;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.Resource;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.stereotype.Service;
+import org.springframework.util.Assert;
+
+import eu.arrowhead.common.exception.ArrowheadException;
+import eu.arrowhead.core.qos.QosMonitorConstants;
+import eu.arrowhead.core.qos.dto.IcmpPingDTOConverter;
+import eu.arrowhead.core.qos.dto.IcmpPingResponse;
+import eu.arrowhead.core.qos.dto.event.monitoringevents.FinishedMonitoringMeasurementEventDTO;
+import eu.arrowhead.core.qos.dto.event.monitoringevents.InterruptedMonitoringMeasurementEventDTO;
+import eu.arrowhead.core.qos.dto.event.monitoringevents.MeasurementMonitoringEvent;
+import eu.arrowhead.core.qos.dto.event.monitoringevents.ReceivedMonitoringRequestEventDTO;
+import eu.arrowhead.core.qos.dto.event.monitoringevents.StartedMonitoringMeasurementEventDTO;
+
+@Service
+public class PingEventProcessor {
+	//=================================================================================================
+	// members
+
+	private static final long SLEEP_PERIOD = TimeUnit.SECONDS.toMillis(1);
+	private static final String SUCCESS_FINISH_PHRASE = " successfully finished ";
+	private static final String INTERRRUPTED_FINISH_PHRASE = " get interrupted";
+	private static final String TIMEOUT_FINISH_PHRASE = " timed out";
+
+	@Resource(name = QosMonitorConstants.EVENT_BUFFER)
+	private ConcurrentHashMap<UUID, PingEventBufferElement> eventBuffer;
+
+	private final Logger logger = LogManager.getLogger(PingEventProcessor.class);
+
+	//=================================================================================================
+	// methods
+
+	//-------------------------------------------------------------------------------------------------	
+	public List<IcmpPingResponse> processEvents(final UUID id, final long timeOut) {
```

Comment:
what is timeOut? Is it a duration as its name suggest (e.g. 10 seconds), or is it a point of time in the future?. Because of the condition of the while loop impies the latter. In the latter case you should check that this point of time is indeed in the future.

Reply:
the use of this method confirms that the timeOut is indeed a point of time in the future. So I suggest to rename it: maybe expiryTime or something.

Reply:
fixed